### ISSUE TITLE
Getting a pod by label fails when serving an archive

### ISSUE THREAD

**Issue by weyfonk** [crust-gather/crust-gather#180]
When an archive created by crust-gather is served using eg. `kubectl-crust_gather serve -a $directory`, getting a pod or its logs by label fails:

For instance:
```
$ kubectl get pods -n cattle-fleet-system -l app=fleet-agent
No resources found in cattle-fleet-system namespace.

# This generates the following crust-gather output:
2024-11-19T08:56:47.008957Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.014260Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.017589Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.020614Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.024984Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.027600Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.031166Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.033542Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
2024-11-19T08:56:47.036526Z ERROR table{table="namespaces/cattle-fleet-system/v1/pod/*.yaml"}:try_from{selector="app=fleet-agent"}: kubectl_crust_gather::gather::selector: error=failed to parse value as expression: '-' at 9..10
```

However, getting the same pod by its name succeeds:
```
$ kubectl get pod -n cattle-fleet-system fleet-agent-0 -o=jsonpath="{range .items[*]}{.metadata.name}{'\t'}{.metadata.labels}{'\n'}"                                                                                              
fleet-agent-0   {"app":"fleet-agent","apps.kubernetes.io/pod-index":"0","controller-revision-hash":"fleet-agent-5dd5d5dd8c","statefulset.kubernetes.io/pod-name":"fleet-agent-0"}
```

**Comment by Danil-Grigorev** [comment]
Seems that the problem is within the regex, which should include “-“: https://github.com/crust-gather/crust-gather/blob/e4b942437160e61bf298e2ecf5c0ddeed01096aa/src/gather/selector.rs#L59

May be enough to replace/extend each `\w` with the set from https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

**Comment by Danil-Grigorev** [comment]
Fixed by #181
